litmus-portal/graphql-server/graph/resolver.go

```diff
@@ -8,17 +8,21 @@ import (
 	"github.com/99designs/gqlgen/graphql"
 	"github.com/litmuschaos/litmus/litmus-portal/graphql-server/graph/generated"
 	"github.com/litmuschaos/litmus/litmus-portal/graphql-server/pkg/authorization"
+	"github.com/litmuschaos/litmus/litmus-portal/graphql-server/pkg/chaoshub"
 )
 
 // This file will not be regenerated automatically.
 //
 // It serves as dependency injection for your app, add any dependencies you require here.
 
-type Resolver struct{}
+type Resolver struct {
+	ChaosHubService chaoshub.Service
```

Comment:
This field does not need to be exported
```suggestion
	chaosHubService chaoshub.Service
```

Reply:
Thinking aloud, this `Resolver` should be used for dependency injection,  we can also initialize the mongo client here and use the same mongo client in all our handlers. Was checking the official documentation and found this here: https://gqlgen.com/getting-started/

`The graph/resolver.go file is where we declare our app’s dependencies, like our database. It gets initialized once in server.go when we create the graph.`

`type Resolver struct{
	todos []*model.Todo
}`

Reply:
For the same approach here as well: https://hasura.io/blog/building-a-graphql-api-with-golang-postgres-and-hasura/

Reply:
It would be unnecessary as we are already storing the individual service instance(for eg. chaosHubService). The service object would have the db instance so no point in storing it separately as our resolvers don't directly use the db instance. We can do initialization like :

```
chaoshub.NewService(dbInstance)
```

If we want to parameterize the dbInstance being passed which is a good idea.

Reply:
Yes, we can do something like that ^^